include/unifex/retry_when.hpp

```diff
@@ -97,7 +97,7 @@ class _trigger_receiver<Source, Func, Receiver, Trigger>::type {
   }
 
   template(typename R = Receiver)
-    (requires is_callable_v<decltype(unifex::set_done), Receiver>)
+    (requires receiver<Receiver>)
```

Comment:
I think this should be using `R` instead of `Receiver` to force lazy evaluation of the `receiver` constraint.